site/content/posts/2023-09-18-multiple-idps-and-identity-transformations.md

```diff
@@ -0,0 +1,129 @@
+excerpt: "With the release of v0.26.0, Pinniped now supports multiple identity providers and identity transformations"
+tags: ['Ryan Richard', 'release']
+---
+
+![multiple seals](https://plus.unsplash.com/premium_photo-1661962912663-49c457dda9ca?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&w=4144&q=80)
+*Photo from [Unsplash](https://unsplash.com/photos/fcmxfsAgxqU)*
+
+Pinniped's v0.26.0 release includes several significant new features described below.
+Additionally, the release includes several dependency updates and fixes.
+See the [release notes](https://github.com/vmware-tanzu/pinniped/releases/tag/v0.26.0) for more details.
+
+## Background: Identity provider resources
+
+In Kubernetes, a user identity is a username string and a list of group name strings.
+
+Each OIDCIdentityProvider, LDAPIdentityProvider, or ActiveDirectoryIdentityProvider configures how to use a specific
+protocol (e.g. OIDC or LDAP) to authenticate users and to extract their identities (usernames and group names) for
+Kubernetes from those external identity providers.
+
+Prior to this release, the Pinniped Supervisor would only allow the use of a single OIDCIdentityProvider, LDAPIdentityProvider,
+or ActiveDirectoryIdentityProvider at a time. With this new release, you can now use multiple simultaneous identity
+providers from which to draw user identities.
+
+## New: Multiple FederationDomains
+
+The Pinniped Supervisor offers a custom resource called FederationDomain.
+Any Kubernetes cluster may trust a Supervisor's FederationDomain to provide authentication services for the cluster.
+Each FederationDomain defines which users may authenticate into those clusters, and how those users go about authenticating.
+So, for a given FederationDomain, the *same users* may authenticate into *all* the clusters which trust that FederationDomain.
+
+Once authenticated, each cluster may use different authorization policies (e.g. Kubernetes RBAC rules) to determine the
+capabilities of each user. Pinniped provides user authentication, while still allowing you to choose your preferred
+system for user authorization on each cluster.
+
+You may configure multiple FederationDomains in the Pinniped Supervisor. Each must have a unique URL called an "issuer URL".
+These URLs can be unique by using different hostnames (a form of virtual hosting) or they can simply have different paths on the same host.
+
+When a user authenticates to a FederationDomain, they start a single sign-on session for that FederationDomain only.
+The user may use all clusters which trust that FederationDomain for the rest of the day without being prompted to
+authenticate again. Behind the scenes, the Supervisor is constantly checking with the external identity provider to ensure
+that the user should be allowed to continue their ongoing session.
```

Comment:
`continue their ongoing session....`. This is valuable because it ensures sessions do not last longer than intended, and users who should no longer access cluster(s) cannot do so.